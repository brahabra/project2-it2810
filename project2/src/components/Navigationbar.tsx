import React from 'react'
import Navbar from 'react-bootstrap/Navbar'
import Nav from 'react-bootstrap/Nav';
import Container from 'react-bootstrap/Container';
import '../styles/Navigationbar.css'
import logo from "../gitlab-logo-400.png"

export default function Navigationbar() {
  return (
    <>

      <nav className="navbar navbar-dark navbar-expand-md bg-background-color justify-content-center">
        <a href="/" className="navbar-brand mr-0">
          <img src={logo} alt="GitLab Logo" />
        </a>
        <div className="justify-content-between align-items-center w-100" >
            <ul className="navbar-nav abs-center-x">
                <li className="nav-item">
                    <a className="nav-link" href="/commits">Commits</a>
                </li>
                <li className="nav-item">
                    <a className="nav-link" href="/issues">Issues</a>
                </li>
            </ul>
        </div>
      </nav>

    </>
  )
}
/*
<>
    <nav className="navbar navbar-dark navbar-expand-md bg-background-color justify-content-center">
      <a href="/" className="navbar-brand mr-0">
        <img src={logo} alt="GitLab Logo" />
      </a>
      <div className="navbar-nav text-center">
          <ul className="navbar-nav mx-auto text-center">
              <li className="nav-item">
                  <a className="nav-link" href="/commits">Commits</a>
              </li>
              <li className="nav-item">
                  <a className="nav-link" href="/issues">Issues</a>
              </li>
          </ul>
      </div>
    </nav>
    
    <nav className="navbar navbar-light navbar-expand-md bg-dark justify-content-center">
    <a href="/" className="navbar-brand mr-0">
      <img src={logo} alt="GitLab Logo" />
    </a>
    <div className="justify-content-between align-items-center w-100 ">
        <ul className="navbar-nav mx-auto text-center abs-center-x">
            <li className="nav-item active">
                <a className="nav-link" href="/commits">Commits</a>
            </li>
            <li className="nav-item">
                <a className="nav-link" href="/issues">Issues</a>
            </li>
        </ul>
    </div>
</nav>
</>
*/