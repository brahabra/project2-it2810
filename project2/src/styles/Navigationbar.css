.bg-background-color {
    background-color: #58D68D;
}

img {
    width: 150px;
}

.abs-center-x {
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
}