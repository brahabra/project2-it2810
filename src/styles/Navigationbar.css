/* Css for navbar (lots of the styling happens in styles.tsx, because we used material ui.
material ui components cannot directly be styled in css.) */

.bg-background-color {
    background-color: #58D68D;
}

.navbar-logo {
    width: 150px;
    cursor: pointer;
}

@media only screen and (max-width: 600px) {
    .navbar-logo {
        visibility: hidden;
    }
}

.abs-center-nav-link {
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
    bottom: 20%;
}