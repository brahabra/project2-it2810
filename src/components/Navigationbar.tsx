import "../styles/Navigationbar.css";
import logo from "../img/gitlab-logo-400.png";
import { Button } from "@mui/material";
import { style } from "../styles/Styles";

export enum Page {
  Home,
  Commits,
  Issues,
}

//Properties for navbar component
interface Props {
  page: Page;
  checkAndSetPage: (value: Page) => void;
}

//Navigationbar component.
export default function Navigationbar(props: Props) {
  return (
    <nav className="navbar navbar-dark navbar-expand bg-background-color">
      <img
        className="navbar-logo"
        src={logo}
        alt="GitLab Logo"
        onClick={() => props.checkAndSetPage(Page.Home)}
      />
      <div className="align-items-center">
        <ul className="navbar-nav abs-center-nav-link">
          <li className="nav-item">
            <Button
              sx={style.navbarButton}
              onClick={() => props.checkAndSetPage(Page.Home)}
            >
              Home
            </Button>
          </li>
          <li className="nav-item">
            <Button
              sx={style.navbarButton}
              onClick={() => props.checkAndSetPage(Page.Commits)}
            >
              Commits
            </Button>
          </li>
          <li className="nav-item">
            <Button
              sx={style.navbarButton}
              onClick={() => props.checkAndSetPage(Page.Issues)}
            >
              Issues
            </Button>
          </li>
        </ul>
      </div>
    </nav>
  );
}
