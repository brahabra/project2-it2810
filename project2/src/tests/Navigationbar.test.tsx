import * as React from 'react'
import Navigationbar from '../components/Navigationbar'

const ReactTestRenderer = require('react-test-renderer');

it('renders correctly', () => {
  const tree = ReactTestRenderer.create(<Navigationbar/>).toJSON()
  expect(tree).toMatchSnapshot()
})