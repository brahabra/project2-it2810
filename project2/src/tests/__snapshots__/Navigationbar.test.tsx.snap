// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-background-color"
  style={
    {
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <a
    className="navbar-brand"
    href="/"
  >
    Logo
  </a>
  <div
    className="navbar-nav"
    onKeyDown={[Function]}
  >
    <a
      className="nav-link"
      data-rr-ui-event-key="/commits"
      disabled={false}
      href="/commits"
      onClick={[Function]}
    >
      Commits
    </a>
    <a
      className="nav-link"
      data-rr-ui-event-key="/issues"
      disabled={false}
      href="/issues"
      onClick={[Function]}
    >
      Issues
    </a>
  </div>
</nav>
`;
